### PR TITLE
Added pagination for projects and some new tweaks

### PR DESCRIPTION
**Pull request by JiffsMaverick**
Hello! Thanks for this awesome project. It was very helpful. This pull request add some features to this project:

* Pagination for projects
* More customization for actions with project, badges, webhooks, descriptions
* Special variable for curl to additional options (for example: key "-k" for insecure connections)
* Note about requirements in README

### CODE REVIEW

File: migrate.sh [hunk 1]
```diff
@@ -35,30 +54,63 @@ function urlencode() {
...
   local -a projects
-  mapfile -t projects < <(curl -sS --header "${authHeaderSourceGitlab}" "${groupProjectsUrl}" | jq -r '.[].path_with_namespace')
+  mapfile -t projects <<< "$(getObjects)"
+
   local -a archivedProjects
-  mapfile -t archivedProjects < <(curl -sS --header "${authHeaderSourceGitlab}" "${groupProjectsUrl}&archived=true" | jq -r '.[].path_with_namespace')
+  mapfile -t archivedProjects <<< "$(getObjects "archived")"
+
+  if [[ "${#archivedProjects[@]}" == 1 ]]; then
+    archivedProjects=()
```
Comment (albuch) [inline migrate.sh:104]: Could you elaborate why this condition exists?

Comment (JiffsMaverick) [reply migrate.sh:104]: Oh, I'm sorry for this kludge. I get empty array without it, if I don't have archived projects. Example output without it:

```
Migrating project '': <-- sic!
	Skipping already existing target project.
```

I haven't thought of anything better :(

**Comment by JiffsMaverick** [comment]
Ok, done!

File: migrate.sh (hunk 1 above)
Comment (albuch) [reply migrate.sh:104]: Thanks, I'll have a look into this once I find the time. There should be a more clean solution for your issue.

**Pull request merged**